Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2021
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Utils/WarpXAlgorithmSelection.H"
+#include "FiniteDifferenceSolver.H"
+#include "Utils/WarpXConst.H"
+#include <AMReX_Gpu.H>
+
+using namespace amrex;
+
+/**
+ * \brief Update the E field at the boundary, using the Silver-Mueller condition
+ */
+void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 >& Efield,
+    std::array< std::unique_ptr<amrex::MultiFab>, 3 > const& Bfield,
+    amrex::Box domain_box,
+    amrex::Real const dt ) {
+
+#ifdef WARPX_DIM_RZ
+    amrex::Abort("The Silver-Mueller boundary conditions cannot be used in RZ geometry.");
+#else
+
+    // Ensure that we are using the Yee solver
+    if (m_fdtd_algo != MaxwellSolverAlgo::Yee) {
+        amrex::Abort("The Silver-Mueller boundary conditions can only be used with the Yee solver.");
+    }
+
+    // Ensure that we are using the cells the domain
+    domain_box.enclosedCells();
+
+    // Calculate relevant coefficients
+    amrex::Real const cdt_over_dx = PhysConst::c*dt*m_stencil_coefs_x[0];
+    amrex::Real coef1_x = (1. - cdt_over_dx)/(1. + cdt_over_dx);
+    amrex::Real coef2_x = 2*cdt_over_dx/(1. + cdt_over_dx) / PhysConst::c;
+#ifdef WARPX_DIM_3D
+    amrex::Real const cdt_over_dy = PhysConst::c*dt*m_stencil_coefs_y[0];
+    amrex::Real coef1_y = (1. - cdt_over_dy)/(1. + cdt_over_dx);
+    amrex::Real coef2_y = 2*cdt_over_dy/(1. + cdt_over_dy) / PhysConst::c;
+#endif
+    amrex::Real const cdt_over_dz = PhysConst::c*dt*m_stencil_coefs_z[0];
+    amrex::Real coef1_z = (1. - cdt_over_dz)/(1. + cdt_over_dx);
+    amrex::Real coef2_z = 2*cdt_over_dz/(1. + cdt_over_dz) / PhysConst::c;
```

Comment:
If you like you can `const`-ify those. Let's also avoid the `int`s and potential `float`/`double` mixes:
```suggestion
    amrex::Real const coef1_x = (1._rt - cdt_over_dx)/(1._rt + cdt_over_dx);
    amrex::Real const coef2_x = 2._rt*cdt_over_dx/(1._rt + cdt_over_dx) / PhysConst::c;
#ifdef WARPX_DIM_3D
    amrex::Real const cdt_over_dy = PhysConst::c*dt*m_stencil_coefs_y[0];
    amrex::Real const coef1_y = (1._rt - cdt_over_dy)/(1._rt + cdt_over_dx);
    amrex::Real const coef2_y = 2._rt*cdt_over_dy/(1._rt + cdt_over_dy) / PhysConst::c;
#endif
    amrex::Real const cdt_over_dz = PhysConst::c*dt*m_stencil_coefs_z[0];
    amrex::Real const coef1_z = (1._rt - cdt_over_dz)/(1._rt + cdt_over_dx);
    amrex::Real const coef2_z = 2._rt*cdt_over_dz/(1._rt + cdt_over_dz) / PhysConst::c;
```